### PR TITLE
feat: Add Unleash Provider

### PR DESCRIPTION
**Pull request by liran2000**
Add [Unleash](https://www.getunleash.io/) Provider.

See [readme](https://github.com/liran2000/go-sdk-contrib/blob/55ec211520c75761b34abafee762bbbb1be8b856/providers/unleash/README.md#unofficial-unleash-openfeature-go-provider) for details.

@sighphyre @gastonfournier @ivarconr you are welcome to review from Unleash perspective.

@Kavindu-Dodan @toddbaert you are welcome to review from OpenFeature perspective.

### CODE REVIEW

**Comment by ivarconr** [comment]
nice 👍🏼

File: providers/unleash/pkg/provider.go
Comment (Kavindu-Dodan) [inline providers/unleash/pkg/provider.go:17]: Given that there's no event stream usage, this can be safely removed (including `EventChannel` implementation)

From OF point of view, it is not essential for a provider to support events :)

File: providers/unleash/pkg/provider.go
Comment (Kavindu-Dodan) [inline providers/unleash/pkg/provider.go:38]: We can complete the `StateHandler` interface implementation by implementing `Init(evaluationContext EvaluationContext) error`

And the initialization can wrap the `unleash.Initialize(...)` , including handling errors.

**Review by Kavindu-Dodan — APPROVED** [review]
Overl looks good from OF point of view. Please consider the review commits and related implementations. 

@toddbaert hope you can have a look as well

File: providers/unleash/README.md
Comment (sighphyre) [inline providers/unleash/README.md:46]:
```suggestion
	resolution := provider.StringEvaluation(ctx, "variant-flag", "", nil)
```

File: providers/unleash/pkg/provider.go
Comment (sighphyre) [inline providers/unleash/pkg/provider.go:26]: I think the lint is correct here. Initialize can fail, some of these are hard error cases like a broken URL, so we probably need to handle this somehow

File: providers/unleash/pkg/provider.go
Comment (sighphyre) [inline providers/unleash/pkg/provider.go:147]: From a purely Unleash perspective, this is a variant, not a feature. If it makes more sense within this context to use `feature`, that's totally fine, but renaming this variable to variant makes a lot more sense to my eyes

Comment (sighphyre) [inline providers/unleash/pkg/provider.go:147]: I don't think we're passing the Unleash context here, which is definitely needed for some toggles to successfully evaluate

File: providers/unleash/pkg/provider.go
Comment (sighphyre) [inline providers/unleash/pkg/provider.go:84]: Same as below, I think this needs to be passed the evalCtx parameter

File: providers/unleash/pkg/provider_test.go
Comment (sighphyre) [inline providers/unleash/pkg/provider_test.go:1]: I'd really like to see some tests that check that the context passed in is respected. In the test data you have a "users-flag", which would need a context with userid on it to successfully return true. That might be a nice flag to build some tests off of for this

**Comment by toddbaert** [comment]
I think you might have force pushed something in error, @liran2000, I'm not seeing the changed files referenced so far in the review.

I'll check back again later.

File: providers/unleash/pkg/provider.go
Comment (toddbaert) [inline providers/unleash/pkg/provider.go:33]: I think we'd want to set the state to error if `err == nil`.

File: providers/unleash/pkg/provider.go
Comment (liran2000) [reply providers/unleash/pkg/provider.go]: Thanks, removed it.
Unleash [supports events listener](https://github.com/Unleash/unleash-client-go/blob/2b58cc4de44e179461a22a29660f049d24d50d4c/unleash_test.go#L18), if you want to take it for a separate PR, by something like converting the unleash events to OpenFeature events, we can discuss separately.

File: providers/unleash/pkg/provider.go
Comment (toddbaert) [inline providers/unleash/pkg/provider.go:198]: May want to use "Invalid Context" error here: https://github.com/open-feature/go-sdk/blob/d8da306f06a5159abdb088b91e78fb59cb0b68a5/pkg/openfeature/resolution_error.go#L19

File: providers/unleash/pkg/provider_test.go
Comment (liran2000) [reply providers/unleash/pkg/provider_test.go:1]: [Added](https://github.com/open-feature/go-sdk-contrib/pull/338/files#diff-1cb45209a551a2fbaa8ebc0e5d4be5ae1f5e3381048e165e0ff15c22034b21c4R143).

**Review by toddbaert — APPROVED** [review]
LGTM, but I will await approval from one Unleashed contributor before merge!

**Comment by liran2000** [comment]
@sighphyre @Kavindu-Dodan @toddbaert thank you for the valuable feedback, I addressed your comments.

* had to rebase due to missing _sign-off_ in some commits from a different IDE.

File: .release-please-manifest.json
Comment (toddbaert) [inline .release-please-manifest.json:12]: @liran2000 I guess this was a copy paste error, and you mean to do `providers/unleash`. I will force a quick fix for this, right now this is causing a 1.0 release which I don't think you want.

Sorry I missed this! :sweat_smile:

Comment (toddbaert) [reply .release-please-manifest.json:12]: Fixed and release looks good now: https://github.com/open-feature/go-sdk-contrib/pull/346

**Pull request merged**
